File: Python/String-2/get_sandwich.py

# a sandwich is two pieces of bread with something in between
# return the string that is between the first and last appearance of "bread" in the given string, or return the empty string "" if there are not two pieces of bread
def get_sandwich(str):
    s = ''
    front = 0
    back = 0
    found = False
    bread = 0

    i = 0
    while (i < len(str)-4 and not found):
        if str[i:i+5] == 'bread':
            front = i+5
            found = True
        i+=1
    
    found = False
    i = len(str)
    while (i >= 5 and not found):
        if str[i-5:i] == 'bread':
            back = i-5
            found = True
        i-=1
    
    for x in range(len(str)-4):
        if str[x:x+5] == 'bread':
            bread+=1

    if bread > 1:
        s = s + str[front:back]

    return s

print(get_sandwich('breadjambread'))
print(get_sandwich('xxbreadjambreadyy'))
print(get_sandwich('xxbreadyy'))

File: Java/String-2/getSandwich.java

public class string2 {
	
	// a sandwich is two pieces of bread with something in between
    // return the string that is between the first and last appearance of "bread" in the given string, or return the empty string "" if there are not two pieces of bread
	public String getSandwich(String str) {
		
		String s = "";
		int front = 0;
		int back = 0;
		boolean found = false;
		int bread = 0;
		
		for (int i = 0; i < str.length()-4 && !found; i ++) {
			if (str.substring(i, i+5).equals("bread")) {
				front = i+5;
				found = true;
			}
		}
		found = false;
		for (int i = str.length(); i >= 5 && !found; i--) {
			if (str.substring(i-5, i).equals("bread")) {
				back = i-5;
				found = true;
			}
		}
		
		for (int i = 0; i < str.length()-4; i++) {
			if (str.substring(i, i+5).equals("bread")) {
				bread++;
			}
		}
		
		if (bread > 1) {
			s = s + str.substring(front, back);
		}
		
		return s;
		
	}

	public static void main(String[] args) {

		string2 o = new string2();
		
		System.out.println(o.getSandwich("breadjambread"));
		System.out.println(o.getSandwich("xxbreadjambreadyy"));
		System.out.println(o.getSandwich("xxbreadyy"));

	}

}
